### PR TITLE
Introduce frustum-based point feature noise.

### PR DESCRIPTION
**Pull request by lengzq**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [ ] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by lengzq** [comment]
@tanzhenyu

File: keras_cv/layers/preprocessing3d/frustum_random_point_feature_noise.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    This layer will randomly select a point from the point cloud as the center of a frustum then generate a frustum based
+    on r_distance, theta_width, and phi_width. Uniformly sampled features noise from [1-max_noise_level, 1+max_noise_level] will be multiplied
+    to points inside the selected frustum. Here, we perturbe point features other than (x, y, z, class).
+    The point_clouds tensor shape must be specific and cannot be dynamic.
+    During inference time, the output will be identical to input. Call the layer with `training=True` to add noise to the input points.
+
+    Input shape:
+      point_clouds: 3D (multi frames) float32 Tensor with shape
+        [num of frames, num of points, num of point features].
+        The first 5 features are [x, y, z, class, range].
```
Comment (tanzhenyu) [inline keras_cv/layers/preprocessing3d/frustum_random_point_feature_noise.py:40]: does `range` here need to be specified? Seems it's adding noise to anything other than x,y,z

**Review by tanzhenyu — APPROVED** [review]

File: keras_cv/layers/preprocessing3d/frustum_random_point_feature_noise.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            raise ValueError("max_noise_level must be >=0 and <=1.")
+
+        self._r_distance = r_distance
+        self._theta_width = theta_width
+        self._phi_width = phi_width
+        self._max_noise_level = max_noise_level
+
+    def get_random_transformation(self, point_clouds, **kwargs):
+        # Randomly select a point from the first frame as the center of the frustum.
+        num_valid_points = tf.cast(
```
Comment (ianstenbit) [inline keras_cv/layers/preprocessing3d/frustum_random_point_feature_noise.py:78]: This looks like a lot of ~duplicate code from the FrustrumRandomDropping layer. 

Maybe we should write a util to build a frustrum and identify which points are in it?

File: keras_cv/layers/preprocessing3d/frustum_random_point_feature_noise_test.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    FrustumRandomPointFeatureNoise,
+)
+
+POINT_CLOUDS = base_augmentation_layer_3d.POINT_CLOUDS
+BOUNDING_BOXES = base_augmentation_layer_3d.BOUNDING_BOXES
+POINTCLOUD_LABEL_INDEX = base_augmentation_layer_3d.POINTCLOUD_LABEL_INDEX
+
+
+class FrustumRandomPointFeatureNoiseTest(tf.test.TestCase):
+    def test_augment_point_clouds_and_bounding_boxes(self):
```
Comment (ianstenbit) [inline keras_cv/layers/preprocessing3d/frustum_random_point_feature_noise_test.py:28]: For this test, could we specify a set of e.g. 10 known points where half are inside and half are outside of the frustum, and then assert that exactly the half that we expect to have noise added are the ones that have noise added?

File: keras_cv/layers/preprocessing3d/frustum_random_point_feature_noise.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        The first 7 features are [x, y, z, dx, dy, dz, phi].
+
+    Output shape:
+      A dictionary of Tensors with the same shape as input Tensors.
+
+    Arguments:
+      r_distance: A float scalar sets the starting distance of a frustum.
+      theta_width: A float scalar sets the theta width of a frustum.
+      phi_width: A float scalar sets the phi width of a frustum.
+      max_noise_level: A float scalar sets the probability threshold for dropping the points.
```
Comment (ianstenbit) [inline keras_cv/layers/preprocessing3d/frustum_random_point_feature_noise.py:52]: Wrong docstring here

File: keras_cv/layers/preprocessing3d/frustum_random_point_feature_noise.py [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    The point_clouds tensor shape must be specific and cannot be dynamic.
+    During inference time, the output will be identical to input. Call the layer with `training=True` to add noise to the input points.
+
+    Input shape:
+      point_clouds: 3D (multi frames) float32 Tensor with shape
+        [num of frames, num of points, num of point features].
+        The first 5 features are [x, y, z, class, range].
+      bounding_boxes: 3D (multi frames) float32 Tensor with shape
+        [num of frames, num of boxes, num of box features].
+        The first 7 features are [x, y, z, dx, dy, dz, phi].
```
Comment (ianstenbit) [inline keras_cv/layers/preprocessing3d/frustum_random_point_feature_noise.py:43]: Can we update this to reference the format like the other layers now do?

"""
Boxes are expected
        to follow the CENTER_XYZ_DXDYDZ_PHI format. Refer to
        https://github.com/keras-team/keras-cv/blob/master/keras_cv/bounding_box_3d/formats.py
        for more details on supported bounding box formats.
"""

File: keras_cv/layers/preprocessing3d/frustum_random_point_feature_noise.py [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+                point_cloud.within_a_frustum(
+                    point_clouds[f],
+                    randomly_select_frustum_center,
+                    self._r_distance,
+                    self._theta_width,
+                    self._phi_width,
+                )[tf.newaxis, :, tf.newaxis]
+            )
+        frustum_mask = tf.concat(frustum_mask, axis=0)
+        feature_noise = 1 + tf.random.uniform(
```
Comment (ianstenbit) [inline keras_cv/layers/preprocessing3d/frustum_random_point_feature_noise.py:100]: What is the purpose of adding `1` to the noise? Do we expect to move all points in the positive direction in each axis?

**Review by ianstenbit — CHANGES_REQUESTED** [review]
Thanks!

File: keras_cv/layers/preprocessing3d/frustum_random_point_feature_noise.py (hunk 1 above)
Comment (lengzq) [reply keras_cv/layers/preprocessing3d/frustum_random_point_feature_noise.py:40]: Removed range.

File: keras_cv/layers/preprocessing3d/frustum_random_point_feature_noise.py (hunk 2 above)
Comment (lengzq) [reply keras_cv/layers/preprocessing3d/frustum_random_point_feature_noise.py:78]: Good point! We could clean it up in a future PR. The redundant codes are random sample a frustum center. 
Most of the frustum related utils are in this function `point_cloud.within_a_frustum`

File: keras_cv/layers/preprocessing3d/frustum_random_point_feature_noise.py (hunk 5 above)
Comment (lengzq) [reply keras_cv/layers/preprocessing3d/frustum_random_point_feature_noise.py:43]: Done.

File: keras_cv/layers/preprocessing3d/frustum_random_point_feature_noise_test.py (hunk 3 above)
Comment (lengzq) [reply keras_cv/layers/preprocessing3d/frustum_random_point_feature_noise_test.py:28]: Add a new test `test_augment_specific_point_clouds_and_bounding_boxes`

**Review by ianstenbit — APPROVED** [review]
